onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.cc

```diff
@@ -106,13 +144,42 @@ bool HasValidQuantizationScales(const InitializedTensorSet& initializers, const
     }
     const auto scale_name = input_defs[idx]->Name();
     if (Contains(initializers, scale_name)) {
-      const auto& tensor = *initializers.at(scale_name);
-      if (!tensor.dims().empty() && tensor.dims()[0] != 1) {
-        LOGS_DEFAULT(VERBOSE) << op << " does not support per-channel quantization";
-        return false;
+      const auto& scale_tensor = *initializers.at(scale_name);
+      bool is_conv_weight = is_qlinear_conv && idx == 4;
+      bool is_conv_u8s8_weight = false;
+
+      if (is_conv_weight) {
+        const auto& weight_tensor = *initializers.at(node.InputDefs()[3]->Name());
+        is_conv_u8s8_weight = is_conv_weight &&
+                              weight_tensor.data_type() == ONNX_NAMESPACE::TensorProto_DataType_INT8;
+      }
+
+      // We need to check the per-channel quantization scales dimensions for u8s8 QlinearConv
+      // We only support per-channel quantization for u8s8
+      // For all other cases, the scales should be a scalar
+      if (is_conv_u8s8_weight) {
+        if (params.android_sdk_ver < 29) {
+          LOGS_DEFAULT(VERBOSE) << op_type << " does not support per-channel quantization on Android API 28-, "
+                                << "system API level: " << params.android_sdk_ver;
+          return false;
+        }
+
+        int64_t scales_dim = scale_tensor.dims().empty() ? 1 : scale_tensor.dims()[0];
+        const auto& weight_tensor = *initializers.at(node.InputDefs()[3]->Name());
+        if (weight_tensor.dims()[0] != scales_dim) {
+          LOGS_DEFAULT(VERBOSE) << op_type << " mismatch int8 per-channel quantization weight,"
+                                << " weight dimension[0] " << weight_tensor.dims()[0]
+                                << " scale dimension " << scales_dim;
+          return false;
+        }
+      } else {
+        if (!scale_tensor.dims().empty() && scale_tensor.dims()[0] != 1) {
+          LOGS_DEFAULT(VERBOSE) << op_type << " does not support per-channel quantization";
+          return false;
```

Comment:
nit: message should say why. right now it would give the impression there's no support at all for per-channel q10n